sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -222,10 +216,16 @@ public RehydrationToken GetRehydrationToken()
             string finalStateVia,
             string? operationId = null)
         {
-            var data = new BinaryData(new { version = RehydrationTokenVersion, id = operationId, requestMethod = requestMethod.ToString(), initialUri = startRequestUri.AbsoluteUri, nextRequestUri, headerSource, lastKnownLocation, finalStateVia });
+            // TODO: Once we remove NextLinkOperationImplementation from internal shared and make it internal to Azure.Core only, we can access the internal members from RehydrationToken directly
+            var json = $$"""
+            {"version":"{{RehydrationTokenVersion}}","id":{{ConstructStringValue(operationId)}},"requestMethod":"{{requestMethod}}","initialUri":"{{startRequestUri.AbsoluteUri}}","nextRequestUri":"{{nextRequestUri}}","headerSource":"{{headerSource}}","finalStateVia":"{{finalStateVia}}","lastKnownLocation":{{ConstructStringValue(lastKnownLocation)}}}
```

Comment:
This is a lot of work with a non-trivial cost just to create an instance of `RehydrationToken`.   Should we create some kind of factory method or just a public constructor for this?   

Given the very specific scope where `RehydrationToken` is used, I wonder if the hiding the constructor from customers to avoid potential confusion justifies the serialization cost.

//cc: @KrzysztofCwalina, @annelo-msft

Reply:
`RehydrationToken` is only constructed internally for now, that's why the members are internal.
And only if we want to access the internal members in Azure.Core shared code like in NextLinkOperationImplementataion, we need to have this workarounds.